src/main/java/com/faforever/client/leaderboard/LeaderboardController.java

```diff
@@ -120,6 +137,28 @@ protected void onDisplay(NavigateEvent navigateEvent) {
     });
   }
 
+
+  public void onContextMenuRequested(ContextMenuEvent contextMenuEvent, TableRow<LeaderboardEntry> leaderboardEntryTableRow) {
+
+    if (contextMenuController != null) {
+      LeaderboardUserContextMenuController controller = contextMenuController.get();
+      if (controller != null) {
+        controller.getContextMenu().show(leaderboardEntryTableRow, contextMenuEvent.getScreenX(), contextMenuEvent.getScreenY());
+        return;
+      }
+    }
+
+    String playerId = leaderboardEntryTableRow.getItem().getId();
+
+    LeaderboardUserContextMenuController controller = uiService.loadFxml("theme/leaderboard/leaderboard_user_context_menu.fxml");
+
+    leaderboardService.getPlayerObjectsById(playerId).thenAccept(players -> Platform.runLater(() -> {
```

Comment:
Make a request /data/player/id instead of this /data/player?filter=id==id

Reply:
Maybe just refactor this call so it calls the /data/player/id route

Reply:
I created new methods to implement the new route, but the problem remains the same: the ID of LeaderboardEntry and Player simply do not match, therefore leading to NullPointers or wrong contextMenus.